docker-compose.yml

```diff
@@ -114,10 +114,31 @@ services:
     build:
       context: .
       dockerfile: cpp/Dockerfile
-    environment:
-      PARQUET_TEST_DATA: /arrow/cpp/submodules/parquet-testing/data
     volumes: *ubuntu-volumes
 
+  cpp-ci:
+    # Usage:
+    #   docker-compose build cpp-ci
+    #   docker-compose run cpp-ci
+    image: ursalab/arrow:cpp-ci
+    shm_size: 2G
+    cap_add:
+      # LeakSanitizer and gdb requires ptrace(2)
+      - SYS_PTRACE
+    build:
+      context: .
+      dockerfile: cpp/Dockerfile.ubuntu-bionic
+      args:
+        LLVM_VERSION: 7
+    environment:
+      ARROW_FLIGHT: "OFF"
+      ARROW_USE_ASAN: "ON"
+      ARROW_USE_UBSAN: "ON"
+    volumes:
+      - .:/arrow:delegated
```

Comment:
@kszucs can you advise how to use yaml references with an extra entry? I initially tried:
```
volumes: *ubuntu-volumes
    - ${HOME}/.ccache:/build/ccache:cached

Reply:
I was also failed to merge yaml lists, so we should stick with the repetitions.